packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -17,6 +17,7 @@ import {
     isTrue,
     isString,
     StringToLowerCase,
+    APIVersion,
```

Comment:
We should be using [isAPIFeatureEnabled](https://github.com/salesforce/lwc/blob/94432b279559412effb29e3c1b9d7fb1ead8742d/packages/%40lwc/shared/src/api-version.ts#L77) and its abstractions rather than directly checking the version here.